src/Drips.sol

```diff
@@ -705,6 +738,35 @@ library Drips {
         return prev.config.lt(next.config);
     }
 
+    /// @notice Calculates the amount dripped over a time range.
+    /// @param cycleSecs The cycle length in seconds.
+    /// Must be the same in all calls working on a single storage instance. Must be higher than 1.
+    /// @param amtPerSec The dripping rate
+    /// @param start The dripping start time
+    /// @param end The dripping end time
+    /// @param amt The dripped amount
+    function _drippedAmt(
+        uint256 cycleSecs,
+        uint256 amtPerSec,
+        uint256 start,
+        uint256 end
```

Comment:
Yes and it comes from a very dangerous trap. For types smaller than 256 bits Solidity sometimes leaves junk in the higher bytes to save gas on clearing them. It only does the clean-up when it's about to perform an operation which can be affected by the higher bits and opening an inline assembly block isn't considered one of them. This behavior can randomly derail Yul, because it's not aware of the dirtiness of Solidity-managed variables. An explicit conversion to `uint256` forces Solidity to do the clean-ups wherever needed.

Reply:
Ah interesting.